File: week03/채서연/SWEA_26943_수위계의이진눈금.py
T = int(input())
for tc in range(1, T+1):
    n = float(input())
    result = ''
    for i in range(13):
        if i == 12:
            result = 'overflow'
            break
        n *= 2
        result += str(int(n))
        n -= int(n)
        if n == 0:
            break

    print(f'#{tc} {result}')

File: week03/박정수/SWEA_26943_수위계의이진눈금.java
import java.io.*;
import java.util.*;

public class Solution {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int T = Integer.parseInt(br.readLine());

        for (int test_case = 1; test_case <= T; test_case++) {
            double target = Double.parseDouble(br.readLine());

            StringBuilder result = new StringBuilder();

            double h = 0.0;
            double cur = 1;
            boolean find = false;
            for (int i = 0; i < 12; i++) {
                cur /= 2;
                if (target >= h + cur) {
                    h += cur;
                    result.append(1);
                } else {
                    result.append(0);
                }
                if (h == target) {
                    find = true;
                    break;
                }
            }

            sb.append("#").append(test_case).append(" ").append(find ? result : "overflow").append("\n");
        }

        System.out.println(sb);
    }
}
